phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -927,7 +928,16 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
             boolean isViewIndex = TRUE_BYTES_AS_STRING
                     .equals(tableProps.get(MetaDataUtil.IS_VIEW_INDEX_TABLE_PROP_NAME));
 
-            if ((tableType == PTableType.INDEX && !isTransactional) || isViewIndex) {
+            boolean isViewBaseTransactional = false;
+            if (!isTransactional && isViewIndex) {
+                if (tableProps.containsKey(TRANSACTIONAL) &&
+                        Boolean.TRUE.equals(tableProps.get(TRANSACTIONAL))) {
+                    isViewBaseTransactional = true;
```

Comment:
The whole logic is a mess which should be cleaned. I don't understand why we need to check "Boolean.TRUE.equals(tableProps.getOrDefault(TRANSACTIONAL, FALSE))" particularly for viewIndex? Shouldn't "isTransactional = (provider != null) && Boolean.TRUE.equals(tableProps.getOrDefault(TRANSACTIONAL, FALSE))" be applied to both INDEX and viewIndex, and simplify "if (!isTransactional && !isViewBaseTransactional && (tableType == PTableType.INDEX || isViewIndex))" as (!isTransactional && (tableType == PTableType.INDEX || isViewIndex))?